iphone/Classes/CalendarModule.m

```diff
@@ -222,21 +222,45 @@ - (void)requestAuthorization:(JSValue *)callback forEntityType:(EKEntityType)ent
-                                   NSDictionary *propertiesDict;
-                                   if (error == nil) {
-                                     NSString *errorMsg = granted ? nil : @"The user has denied access to events in Calendar.";
-                                     propertiesDict = [TiUtils dictionaryWithCode:(granted ? 0 : 1) message:errorMsg];
-                                   } else {
-                                     propertiesDict = [TiUtils dictionaryWithCode:[error code] message:[TiUtils messageFromError:error]];
-                                   }
-                                   TiThreadPerformOnMainThread(
-                                       ^{
-                                         [callback callWithArguments:@[ propertiesDict ]];
-                                       },
-                                       [NSThread isMainThread]);
-                                 }];
+
+        if (@available(iOS 17.0, *)) {
+
+          NSLog(@"[INFO] Calendar requestAuthorization iOS 17...");
+
+          [ourstore requestFullAccessToEventsWithCompletion:^(BOOL granted, NSError *error) {
+            NSDictionary *propertiesDict;
+            if (error == nil) {
+              NSString *errorMsg = granted ? nil : @"The user has denied access to events in Calendar.";
+              propertiesDict = [TiUtils dictionaryWithCode:(granted ? 0 : 1) message:errorMsg];
+            } else {
+              propertiesDict = [TiUtils dictionaryWithCode:[error code] message:[TiUtils messageFromError:error]];
+            }
+            TiThreadPerformOnMainThread(
+                ^{
+                  [callback callWithArguments:@[ propertiesDict ]];
+                },
+                [NSThread isMainThread]);
+          }];
+        } else {
+
+          NSLog(@"[INFO] Calendar requestAuthorization iOS 16...");
+
+          [ourstore requestAccessToEntityType:EKEntityTypeEvent
+                                   completion:^(BOOL granted, NSError *error) {
+                                     NSDictionary *propertiesDict;
+                                     if (error == nil) {
+                                       NSString *errorMsg = granted ? nil : @"The user has denied access to events in Calendar.";
+                                       propertiesDict = [TiUtils dictionaryWithCode:(granted ? 0 : 1) message:errorMsg];
+                                     } else {
+                                       propertiesDict = [TiUtils dictionaryWithCode:[error code] message:[TiUtils messageFromError:error]];
+                                     }
+                                     TiThreadPerformOnMainThread(
+                                         ^{
+                                           [callback callWithArguments:@[ propertiesDict ]];
+                                         },
+                                         [NSThread isMainThread]);
```

Comment:
It looks like the logic in the completion handler is reusable, so the block should be moved out of the handler and be called per instance. That way, the if-check if also better to read.

Reply:
Thank you for the feedback - I have made the changes to remove the logging and moved the logic for each block into a common area - please check